### ISSUE TITLE
Define & Document a function name prefix for core us in plugins

### ISSUE THREAD

**Issue by Athanasius** [EDCD/EDMarketConnector#1226]
i.e. we have `journal_entry()` now and a proposed `journal_entry_cqc()` (see #1030 ).  But what if a plugin developer already has that function name in their code, with matching number of arguments?

We need to define a prefix for function names that plugin developers know not to use, and then always use it ourselves for any new API function names.

**Comment by inorton** [comment]
For a long time now I've thought that the plugin API entrypoint/discovery should be different. I'd like to see it become class based where plugins inherit from a base class we provide and override methods.

**Comment by Athanasius** [comment]
> For a long time now I've thought that the plugin API entrypoint/discovery should be different. I'd like to see it become class based where plugins inherit from a base class we provide and override methods.

Yes, see @A-UNDERSCORE-D 's work on a wholly new plugin API/paradigm (but with backwards compatibility).

*This* issue is more about avoiding an issue in the time before we get that into a mergable state, and would still apply during the backwards compatibility stage.
